### DISCUSSION TITLE
How many nodes can k3s support?

### DISCUSSION THREAD

**Discussion by PeaceRebel** [k3s-io/k3s#6463] [Q&A]
Given that k3s is meant to be used for edge computing, how many nodes can it support? I want to use k3s for managing around 10K devices(with k3s agent only) and 100K pods. What resources would I need to run k3s server to manages all this?

**Comment by brandond** [comment]
I would probably consult the upstream documentation at https://kubernetes.io/docs/setup/best-practices/cluster-large/ - note that your proposed device count already exceeds the upstream guidelines. You're probably going to have to split things up a bit, in addition to throwing a large amount of hardware at the control-plane.

**Reply by PeaceRebel** [reply]
For IoT applications isn't that number a little too low? k3s doesn't give any improvement over k8s in this aspect? What IoT use case does k3s targets then?

**Reply by brandond** [reply]
https://github.com/k3s-io/k3s/issues/2278#issuecomment-735934710
> FWIW, when we talk about edge we usually mean things like cell towers, stores, restaurants, forward operating bases, etc.

Common edge/iot models using Kubernetes usually involve many small independent clusters, managed by a multi-cluster management tool like rancher and/or fleet. K3s is perfect for running these small clusters on resource constrained nodes. Edge to us does NOT mean one giant cluster with tens of thousands of small nodes. That is sort of the opposite of the way we see the market going.
